Translate Integrating with Other Libraries (#119)
reactjs/id.reactjs.org: content/docs/integrating-with-other-libraries.md

---
id: integrating-with-other-libraries

> title: Integrasi dengan library lain

Reviewer: ```suggestion
title: Integrasi dengan _Library_ Lain
```

> title: Integrasi dengan _Library_ Lain

Reviewer 2: ```suggestion
title: Integrasi dengan Library Lain
```

The `title` over here will not be parsed as Markdown, so let's keep it raw.

permalink: docs/integrating-with-other-libraries.html
---

React dapat digunakan pada web aplikasi apapun. Ini dapat ditanamkan pada aplikasi lain dan, dengan sedikit pemeliharaan, aplikasi lain dapat ditanamkan di React. Panduan ini akan memeriksa beberapa kasus penggunaan yang lebih umum, memfokuskan pada integrasi dengan [jQuery](https://jquery.com/) dan [Backbone](https://backbonejs.org/), tetapi ide yang sama dapat diterapkan untuk mengintegrasikan komponen dengan kode yang ada.

## Integrasi dengan Plugin Manipulasi DOM {#integrating-with-dom-manipulation-plugins}

React tidak akan menyadari perubahan yang dilakukan pada DOM diluar dari React. Ini menentukan pembaharuan berdasarkan representasi internal sendiri, dan jika node DOM yang sama dimanipulasi oleh *library* lain, React menjadi bingung dan tidak memiliki cara untuk memulihkan.

Ini tidak berarti tidak mungkin atau bahkan sulit untuk menggabungkan React dengan cara-cara lain untuk mempengaruhi DOM, Anda hanya perlu memperhatikan apa yang masing-masing lakukan.

> Cara termudah untuk menghindari konflik adalah mencegah komponen React memperbarui. Kamu dapat melakukannya dengan merender elemen-elemen yang React tidak punya alasan untuk diupdate, seperti sebuah `<div />` kosong.

Reviewer 2: Also please replace "kamu" to "Anda".

> ### Bagaimana cara mendekati masalah {#how-to-approach-the-problem}

Reviewer 2: I think we can remove the "bagaimana":

```suggestion
### Cara Pendekatan Masalah {#how-to-approach-the-problem}
```

> Untuk mendemonstrasikan ini, Ayo kita membuat sketsa pembungkus untuk sebuah plugin generik jQuery.

Reviewer 2: I think this is more coherent translation:

```suggestion
Untuk mendemonstrasikan ini, mari kita menggambarkan sebuah pembungkus untuk sebuah plugin generik jQuery.
```

Kita akan melampirkan [ref](/docs/refs-and-the-dom.html) kepada akar elemen DOM. Didalam `componentDidMount`, kita akan mendapat sebuah referensi kepada itu sehingga kita dapat menyampaikan itu kepada plugin jQuery.

> Untuk mencegah React dari menyentuh DOM setelah pemasangan, kita akan mengembalikan sebuah `<div />` kososng dari `render()` *method*.`<div />` elemen tidak memiliki properti atau *children*, sehingga React tidak memiliki alasan untuk memperbaruinya, meninggalkan plugin jQuery bebas untuk mengekelola bagian itu dari DOM:

Reviewer: can we translate *`method`* to be `metode`?

> Untuk mencegah React dari menyentuh DOM setelah pemasangan, kita akan mengembalikan sebuah `<div />` kososng dari `render()` *method*.`<div />` elemen tidak memiliki properti atau *children*, sehingga React tidak memiliki alasan untuk memperbaruinya, meninggalkan plugin jQuery bebas untuk mengekelola bagian itu dari DOM:

Author: @vferdiansyah but from what i have seen in [Glossary](https://github.com/reactjs/id.reactjs.org/wiki/Glosarium) we didn't need to translate method instead we just need to put italic text style to it

> Untuk mencegah React dari menyentuh DOM setelah pemasangan, kita akan mengembalikan sebuah `<div />` kososng dari `render()` *method*.`<div />` elemen tidak memiliki properti atau *children*, sehingga React tidak memiliki alasan untuk memperbaruinya, meninggalkan plugin jQuery bebas untuk mengekelola bagian itu dari DOM:

Reviewer 2: @vferdiansyah We don't translate method, as per the glossary.

}
```

Perhatikan bahwa kami mendefinisikan keduanya `componentDidMount` dan `componentWillUnmount` [lifecycle methods](/docs/react-component.html#the-component-lifecycle). Banyak plugin jQuery melampirkan pendengar event pada DOM sehingga ini penting untuk melepaskan mereka dalam `componentWillUnmount`. Jika plugin tidak menyediakan sebuah *method* untuk membersihkan, kamu mungkin akan menyediakannya sendiri, ingat untuk menghapus pendengar event apapun plugin didaftarkan untuk mencegah kebocoran memori.

### Integrasi dengan Plugin Chosen jQuery {#integrating-with-jquery-chosen-plugin}

Untuk contoh yang lebih konkret dari konsep-konsep ini, mari kita tulis pembungkus minimal untuk plugin [Chosen](https://harvesthq.github.io/chosen/), yang menambah masukan `<select>`.

>**Catatan:**
>
>Hanya karena itu mungkin, bukan berarti itu adalah pendekatan terbaik untuk aplikasi React. Kami menganjurkan kamu untuk menggunakan komponen React saat kamu bisa. komponen React lebih mudah untuk digunakan ulang dalam aplikasi React, dan sering kali menyediakan kontrol yang lebih pada perilaku dan penampilannya.

> Pertama-tama, mari kita lihat pada apa yang Chosen lakukan pada DOM.

Reviewer: ```suggestion
Pertama-tama, mari kita lihat apa yang Chosen lakukan pada DOM.
```

}
```

> Lihat bagaimana kita membungkus `<select>` dalam sebuah ekstra `<div>`. Ini penting karena Chosen akan menambahkan DOM elemen lainnya tepat setelah `<select>` node yang kita berikan pada itu. Namun, sejauh yang React ketahui, `<div>` hanya selalu memiliki satu *children*. Ini adalah bagaimana kita memastikan update React tidak akan konflik dengan DOM node ekstra yang ditambahkan Chosen. Ini juga penting bahwa jika kamu memodifikasi DOM diluar alur React, kamu harus memastikan React tidak punya alasan untuk menyentuh DOM node tersebut.

Reviewer 3: ```suggestion
Lihat bagaimana kita membungkus `<select>` dalam sebuah `<div>` ekstra. Ini penting karena Chosen akan menambahkan elemen DOM lainnya tepat setelah *node* `<select>` yang kita berikan pada itu. Namun, sejauh yang React ketahui, `<div>` hanya selalu memiliki satu *children*. Ini adalah bagaimana kita memastikan pembaharuan React tidak akan konflik dengan ekstra *node* DOM yang ditambahkan Chosen. Ini juga penting bahwa jika kamu memodifikasi DOM di luar alur React, kamu harus memastikan React tidak punya alasan untuk menyentuh *node* DOM tersebut.
```

referensi : https://id.wikipedia.org/wiki/Wikipedia:Daftar_kosakata_bahasa_Indonesia_yang_sering_salah_dieja

}
```

> [**Coba ini di CodePen**](https://codepen.io/gaearon/pen/qmqeQx?editors=0010)

Reviewer: ```suggestion
[**Coba di CodePen**](https://codepen.io/gaearon/pen/qmqeQx?editors=0010)
```

}
```

> [**Coba ini di CodePen**](https://codepen.io/gaearon/pen/bWgbeE?editors=0010)

Reviewer: ```suggestion
[**Coba di CodePen**](https://codepen.io/gaearon/pen/bWgbeE?editors=0010)
```

}
```

> [**Coba ini di CodePen**](https://codepen.io/gaearon/pen/xdgKOz?editors=0010)

Reviewer: ```suggestion
[**Coba di CodePen**](https://codepen.io/gaearon/pen/xdgKOz?editors=0010)
```

> ## Integrasi dengan *Library* Tampilan lainnya {#integrating-with-other-view-libraries}

Reviewer: ```suggestion
## Integrasi dengan *Library* Tampilan Lain {#integrating-with-other-view-libraries}
```

React dapat ditanamkan pada aplikasi-aplikasi lainnya terima kasih pada kefleksiblelannya [`ReactDOM.render()`](/docs/react-dom.html#render).

Meskipun React umum digunakan di *startup* untuk memuat komponen akar React tunggal pada DOM, `ReactDOM.render()` juga bisa dipanggil beberapa kali untuk bagian independen dari UI yang dimana dapat sekecil tombol, atau sebesar sebuah aplikasi.

> Faktanya, inilah tepatnya bagaimana React digunakan di Facebook. Ini membuat kira menulis aplikasi pada React sepotong demi sepotong, dan mengkombinasikannya dengan templat yang dihasilkan server kami dan kode sisi klien lainnya.

Reviewer: ```suggestion
Faktanya, seperti inilah bagaimana React digunakan di Facebook. Ini membuat kita menulis aplikasi pada React sedikit demi sedikit, dan mengkombinasikannya dengan templat yang dihasilkan server kami dan kode sisi klien lainnya.
```

> ### Mengganti Rendering Berbasis String dengan React {#replacing-string-based-rendering-with-react}

Reviewer: ```suggestion
### Mengganti _Rendering_ Berbasis _String_ dengan React {#replacing-string-based-rendering-with-react}
```

> Sebuah pola yang umum di web aplikasi lama adalah untuk mendeskripsikan bingkah dari DOM sebagai string dan memasukannya kepada DOM seperti: `$el.html(htmlString)`. Poin ini dalam sebuah basis kode sempurna untuk memperkenalkan React. Cukup tulis ulang rendering berbasis string sebagai komponen React.

Reviewer 3: ```suggestion
Sebuah pola yang umum pada aplikasi peramban lama adalah untuk mendeskripsikan bingkah dari DOM sebagai *string* dan memasukannya kepada DOM seperti: `$el.html(htmlString)`. Poin ini dalam sebuah basis kode sempurna untuk memperkenalkan React. Cukup tulis ulang *rendering* berbasis *string* sebagai komponen React.
```

> Berikut adalah implementasi jQuery...

Reviewer: ```suggestion
Implementasi jQuery berikut ini...
```

);
```

> [**Coba ini di CodePen**](https://codepen.io/gaearon/pen/RVKbvW?editors=1010)

Reviewer: ```suggestion
[**Coba di CodePen**](https://codepen.io/gaearon/pen/RVKbvW?editors=1010)
```

> Kamu dapat memiliki komponen terisolasi sebanyak yang kamu suka, dan menggunakan `ReactDOM.render()` untuk merendernya pada kontainer DOM yang berbeda. Sedikit demi sedikit, saat anda mengonversi lebih banyak aplikasi ke React, kamu akan bisa mengkombinasikannya kepada komponen yang lebih besar, dan memindahkan beberapa dari `ReactDOM.render()` memanggil hirarki.

Reviewer 3: ```suggestion
Kamu dapat memiliki komponen yang terisolasi sebanyak yang kamu suka, dan menggunakan `ReactDOM.render()` untuk merendernya pada kontainer DOM yang berbeda. Sedikit demi sedikit, saat Anda mengonversi lebih banyak bagian dari aplikasi Anda ke React, kamu akan bisa mengkombinasikannya menjadi komponen yang lebih besar, dan memindahkan beberapa dari hirarki pemanggilan `ReactDOM.render()`.
```

Reviewer: `kamu` -> `Anda`

> ### Menanamkan React didalam sebuah Tampilan Backbone {#embedding-react-in-a-backbone-view}

Reviewer: ```suggestion
### Menanamkan React dalam Tampilan Backbone {#embedding-react-in-a-backbone-view}
```

});
```

> [**Coba ini di CodePen**](https://codepen.io/gaearon/pen/gWgOYL?editors=0010)

Reviewer: ```suggestion
[**Coba di CodePen**](https://codepen.io/gaearon/pen/gWgOYL?editors=0010)
```

Ini juga penting bahwa kita juga dapat memanggil `ReactDOM.unmountComponentAtNode()` pada method `remove` sehingga React membatalkan *event handlers* registrasi dan sumber lainnya yang terkait dengan pohon komponen saat itu dicopot.

Saat sebuah komponen dihapus *dari dalam* sebuah pohon React, pembersihan dilakukan secara otomatis, tapi karena kita menghapus seluruh pohon dengan tangan, kita harus memanggil method ini.

> ## Integrasi degan Lapisan-Lapisan Model {#integrating-with-model-layers}

Reviewer: ```suggestion
## Integrasi dengan Lapisan Model {#integrating-with-model-layers}
```

Meskipun pada umumnya ini dianjurkan untuk menggunakan aliran data searah seperti [React state](/docs/lifting-state-up.html), [Flux](https://facebook.github.io/flux/), atau [Redux](https://redux.js.org/), komopnen React juga dapat menggunakan lapisan model dari kerangkai kerja dan *library* lainnya.

> ### Menggunakan Model Backbone di Komponen-Komponen React {#using-backbone-models-in-react-components}

Reviewer: ```suggestion
### Menggunakan Model Backbone pada Komponen React {#using-backbone-models-in-react-components}
```

Cara termudah untuk menggunakan model [Backbone](https://backbonejs.org/) dan koleksi-koleksi dari sebuah komponen React adalah dengan mendengar pada berbagai macam perubahan *event* dan secara manual memaksakan sebuah perubahan.

Komponen bertanggung jawab untuk merender model-model yang mau mendengar pada *event* `'change'`, Sementara komponen bertanggung jawab untuk merender koleksi-koleksi akan mendengarkan untuk *event* `'add'` dan `'remove'`. Pada kedua kasus tersebut, panggil [`this.forceUpdate()`](/docs/react-component.html#forceupdate) untuk merender ulang komponen dengan data yang baru.

> In the example below, the `List` component renders a Backbone collection, using the `Item` component to render individual items.

Reviewer 2: Please translate this paragraph as well. @mramadhanr

}
```

> [**Coba ini di CodePen**](https://codepen.io/gaearon/pen/GmrREm?editors=0010)

Reviewer: ```suggestion
[**Coba di CodePen**](https://codepen.io/gaearon/pen/GmrREm?editors=0010)
```

);
```

> [**Coba ini di CodePen**](https://codepen.io/gaearon/pen/PmWwwa?editors=0010)

Reviewer: ```suggestion
[**Coba di CodePen**](https://codepen.io/gaearon/pen/PmWwwa?editors=0010)
```
